freqtrade/plugins/pairlist/RemotePairList.py

```diff
@@ -0,0 +1,166 @@
+                          f' {self._pairlist_url}', logger.info)
+
+            if self._keep_pairlist_on_failure:
+                pairlist = self._last_pairlist
+                self.log_once('Keeping last fetched pairlist', logger.info)
+            else:
+                pairlist = []
+
+            time_elapsed = 0
+
+        return pairlist, time_elapsed, info
+
+    def gen_pairlist(self, tickers: Tickers) -> List[str]:
+        """
+        Generate the pairlist
+        :param tickers: Tickers (from exchange.get_tickers). May be cached.
+        :return: List of pairs
+        """
+
+        pairlist = self._pair_cache.get('pairlist')
+        time_elapsed = 0.0
+
+        if pairlist:
+            # Item found - no refresh necessary
+            return pairlist.copy()
+        else:
+            if self._pairlist_url.startswith("file:///"):
+                filename = self._pairlist_url.split("file:///", 1)[1]
+                file_path = Path(filename)
+
+                if file_path.exists():
+                    with open(filename) as json_file:
+                        # Load the JSON data into a dictionary
+                        jsonparse = json.load(json_file)
+                        pairlist = jsonparse['pairs']
+                        info = jsonparse.get('info', '')
+                        self._refresh_period = jsonparse.get('refresh_period', self._refresh_period)
+                        self._pair_cache = TTLCache(maxsize=1, ttl=self._refresh_period)
+
+                else:
+                    raise ValueError(f"{self._pairlist_url} does not exist.")
+            else:
+                # Fetch Pairlist from Remote URL
+                pairlist, time_elapsed, info = self.fetch_pairlist()
+
+        pairlist = self.filter_pairlist(pairlist, tickers)
+        self._pair_cache['pairlist'] = pairlist.copy()
+
+        if time_elapsed != 0.0:
+            self.log_once(f'{info} Fetched in {time_elapsed} seconds.', logger.info)
+        else:
+            self.log_once(f'{info} Fetched Pairlist.', logger.info)
+
+        self._last_pairlist = list(pairlist)
+        return pairlist
+
+    def filter_pairlist(self, pairlist: List[str], tickers: Dict) -> List[str]:
+        """
+        Filters and sorts pairlist and returns the whitelist again.
+        Called on each bot iteration - please use internal caching if necessary
+        :param pairlist: pairlist to filter or sort
+        :param tickers: Tickers (from exchange.get_tickers). May be cached.
+        :return: new whitelist
+        """
+
+        # Validate whitelist to only have active market pairs
+        pairlist = self._whitelist_for_active_markets(pairlist)
+        pairlist = self.verify_blacklist(pairlist, logger.info)
+        # Limit pairlist to the requested number of pairs
+        pairlist = pairlist[:self._number_pairs]
+        self.log_once(f"Searching {self._number_pairs} pairs: {pairlist}", logger.info)
```

Comment:
We should remove this from the pairlists. 
It's quite very misleading (it was misleading in VolumePairlist while it was there) - as it's not reflecting the finally used pairlist.

it's issued at the root level now - once ALL filters have ran - reflecting the actual pairlist that will be used.